web/modules/custom/dpl_event/dpl_event.module

```diff
@@ -173,16 +173,52 @@ function dpl_event_preprocess_field(array &$variables): void {
 
   // After determining the 'event_is_active' flag, check if the event is active.
   if ($variables['event_is_active']) {
-    // If active, set a translatable string as the title.
-    $variables['items'][0]['content']['#title'] = t('Buy tickets');
-    return;
+    $has_free_tickets = FALSE;
+    $categories_fields_to_check = ['event_ticket_categories', 'field_ticket_categories'];
+
+    // Use categories_fields_to_check because the categories field name is
+    // different for event series and instances.
+    foreach ($categories_fields_to_check as $field) {
+      if ($entity->hasField($field)) {
+        $has_free_tickets = dpl_event_has_free_tickets($entity->get($field)
+          ->referencedEntities());
+        if ($has_free_tickets) {
+          break;
+        }
+      }
+    }
+
+    $variables['items'][0]['content']['#title'] = $has_free_tickets ? t('Get tickets') : t('Buy tickets');
   }
   else {
     // Update the title of the button to the label of the current event state.
     $variables['items'][0]['content']['#title'] = $event_state->label();
   }
 }
 
+/**
+ * Check if the event has free tickets.
+ *
+ * @param array<\Drupal\paragraphs\Entity\Paragraph> $ticket_categories
+ *   An array of ticket categories.
+ *
+ * @return bool
+ *   True if the event has free tickets, false otherwise.
+ */
+function dpl_event_has_free_tickets(array $ticket_categories): bool {
```

Comment:
Please consider the signature for your function.

Instead of padding an array of ticket categories you might as well pass an actual event entity. This would also make it easier for others to reuse the function in other cases.